gix-blame/src/types.rs

```diff
@@ -81,60 +101,51 @@ impl BlameRanges {
     /// The range should be 1-based inclusive.
     /// If the new range overlaps with or is adjacent to an existing range,
     /// they will be merged into a single range.
-    pub fn add_range(&mut self, new_range: RangeInclusive<u32>) {
-        self.merge_range(new_range);
+    pub fn add_range(&mut self, new_range: RangeInclusive<u32>) -> Result<(), Error> {
```

Comment:
Can we make the expectation part of the function name, like so: `add_one_based_inclusive_range`?

Same for all other methods that take any kind of `Range`.